lapack/testlapack/general.go

```diff
@@ -1158,29 +1158,51 @@ func extract2x2Block(t []float64, ldt int) (a, b, c, d float64) {
 // isSchurCanonical returns whether the 2×2 matrix [a b; c d] is in Schur
 // canonical form.
 func isSchurCanonical(a, b, c, d float64) bool {
-	return c == 0 || (a == d && math.Signbit(b) != math.Signbit(c))
```

Comment:
I actually find the original easier to read.

`return c == 0 || (b != 0 && a == d && math.Signbit(b) != math.Signbit(c))`

Reply:
You're right. Done.